modules/core/shared/src/main/scala/lucuma/core/syntax/timespan/package.scala

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.core.syntax.timespan
+
+import lucuma.core.util.TimeSpan
+
+extension (inline i: Int) {
+
+  inline def microsecondTimeSpan: TimeSpan =
+    inline if (i >= 0) TimeSpan.fromMicroseconds(i).get
+    else scala.compiletime.error("TimeSpan must be non-negative.")
+
+  inline def µsTimeSpan: TimeSpan =
+    microsecondTimeSpan
+
+  inline def millisecondTimeSpan: TimeSpan =
+    inline if (i >= 0) TimeSpan.fromMilliseconds(BigDecimal(i)).get
+    else scala.compiletime.error("TimeSpan must be non-negative.")
+
+  inline def msTimeSpan: TimeSpan =
+    millisecondTimeSpan
+
+  inline def secondTimeSpan: TimeSpan =
+    inline if (i >= 0) TimeSpan.fromSeconds(BigDecimal(i)).get
+    else scala.compiletime.error("TimeSpan must be non-negative.")
+
+  inline def secTimeSpan: TimeSpan =
+    secondTimeSpan
+
+  inline def minuteTimeSpan: TimeSpan =
+    inline if (i >= 0) TimeSpan.fromMinutes(BigDecimal(i)).get
+    else scala.compiletime.error("TimeSpan must be non-negative.")
+
+  inline def minTimeSpan: TimeSpan =
+    minuteTimeSpan
+
+  inline def hourTimeSpan: TimeSpan =
+    inline if (i >= 0) TimeSpan.fromHours(BigDecimal(i)).get
+    else scala.compiletime.error("TimeSpan must be non-negative.")
+
+  inline def hrTimeSpan: TimeSpan =
+    hourTimeSpan
+
+}
+
+extension (inline l: Long) {
```

Comment:
Tried to generalize this for anything with an `Integral` but couldn't quite swing it with the `inline`s.